doc/devel/matlab_interface_devel.rst

```diff
@@ -4,127 +4,85 @@
-            """ Basic Hello World that displays Hello <name> in MATLAB
-
-            Returns
-            -------
-
-            matlab_output : capture of matlab output which may be
-                            parsed by user to get computation results
+    import os
+    from nipype.interfaces.base import File, traits
+    from nipype.interfaces.matlab import MatlabCommand, MatlabInputSpec
 
-	    Examples
-	    --------
+    class HelloWorldInputSpec( MatlabInputSpec):
+        name = traits.Str(mandatory = True,
+                          desc = 'Name of person to say hello to')
 
-            >>> hello = HelloWorld()
-            >>> hello.inputs.name = 'hello_world'
-            >>> out = hello.run()
-            >>> print(out.outputs.matlab_output)
-            """
-            input_spec = HelloWorldInputSpec
-            output_spec = HelloWorldOutputSpec
+    class HelloWorldOutputSpec(MatlabInputSpec):
+        matlab_output = traits.Str()
 
-	    def _my_script(self):
-        	"""This is where you implement your script"""
-        	script = """
-        	disp('Hello %s Python')
-        	two = 1 + 1
-        	"""%(self.inputs.name)
-        	return script
+    class HelloWorld(MatlabCommand):
+        """ Basic Hello World that displays Hello <name> in MATLAB
 
+        Returns
+        -------
 
-    	    def run(self, **inputs):
-        	## inject your script
-        	self.inputs.script =  self._my_script()
-        	results = super(MatlabCommand, self).run( **inputs)
-        	stdout = results.runtime.stdout
-        	# attach stdout to outputs to access matlab results
-        	results.outputs.matlab_output = stdout
-        	return results
+        matlab_output : capture of matlab output which may be
+                        parsed by user to get computation results
 
+        Examples
+        --------
 
-            def _list_outputs(self):
-                outputs = self._outputs().get()
-                return outputs
+        >>> hello = HelloWorld()
+        >>> hello.inputs.name = 'hello_world'
+        >>> out = hello.run()
+        >>> print(out.outputs.matlab_output)
+        """
+        input_spec = HelloWorldInputSpec
+        output_spec = HelloWorldOutputSpec
 
+        def _my_script(self):
+            """This is where you implement your script"""
+            script = """
+            disp('Hello %s Python')
+            two = 1 + 1
+            """%(self.inputs.name)
+            return script
 
+        def run(self, **inputs):
+            ## inject your script
+            self.inputs.script =  self._my_script()
+            results = super(MatlabCommand, self).run( **inputs)
+            stdout = results.runtime.stdout
+            # attach stdout to outputs to access matlab results
+            results.outputs.matlab_output = stdout
+            return results
 
+        def _list_outputs(self):
+            outputs = self._outputs().get()
+            return outputs
```

Comment:
Lines 29-79 have been superseded by line 82 and should be removed. https://387-59325098-gh.circle-artifacts.com/0/home/circleci/work/docs/devel/matlab_interface_devel.html#example-2